devices/flash-flexspi/fspi/fspi.c

```diff
@@ -52,31 +52,26 @@ __attribute__((section(".noxip"))) static void flexspi_disable(flexspi_t *fspi,
 __attribute__((section(".noxip"))) static inline void flexspi_swreset(flexspi_t *fspi)
 {
 	*(fspi->base + mcr0) |= 1;
-	while ((*(fspi->base + mcr0) & 1) != 0)
-		;
+	while ((*(fspi->base + mcr0) & 1) != 0) {
+	}
 }
 
 
-__attribute__((section(".noxip"))) void flexspi_lutUpdate(flexspi_t *fspi, u32 index, const u32 *lutTable, size_t count)
+__attribute__((section(".noxip"))) static void flexspi_lutSet(flexspi_t *fspi, u32 index, u32 lutCopy[64], size_t count)
 {
-	u32 lutCopy[64];
 	unsigned int i;
 	volatile u32 *lutPtr = fspi->base + lut64 + index;
 
-	for (i = 0; i < count; ++i) {
-		lutCopy[i] = *lutTable++;
-	}
-
 	/* Wait for bus and lut sequencer to be idle */
-	while (((*(fspi->base + sts0) & 2) != 0 && (*(fspi->base + sts0) & 1) != 0) == 0)
-		;
+	while (((*(fspi->base + sts0) & 2) != 0 && (*(fspi->base + sts0) & 1) != 0) == 0) {
+	}
 
 	/* Unlock LUT */
 	*(fspi->base + lutkey) = 0x5af05af0;
 	*(fspi->base + lutcr) = 2;
 
 	/* Update LUT */
-	for (i = 0; i < count; ++i) {
+	for (i = 0; i < count && i < 64; ++i) {
```

Comment:
MISRA
```c
	for (i = 0; (i < count) && (i < 64); ++i) {
```

Reply:
Removed the check (<64), not needed at low level. #DefensiveProgramming